lsr_role2collection.py

```diff
@@ -380,9 +380,11 @@ def run(self):
 
 ALL_DIRS = ROLE_DIRS + PLUGINS + TESTS + DOCS + DO_NOT_COPY
 
-IMPORT_RE = re.compile(br"(\bimport) (ansible\.module_utils\.)(\S+)(.*)$", flags=re.M)
+IMPORT_RE = re.compile(
+    br"(\bimport) (ansible\.module_utils\.)(\S+)(\S*)(\s+#.+|.*)$", flags=re.M
```

Comment:
how does the regexp parser know the difference/boundary between `(\S+)(\S*)` since they both match almost the same thing?

Reply:
Well, if there's no a trailing comment, the last match (match.group(5)) is supposed to be empty. I.e., what I want to have from `(\s+#.+|.*)` is if the line ends with `  # noqa:E501`, set it to match.group(5) otherwise none. Do you think `(\s+#.+|.*)` is a reasonable regex for the purpose?

Reply:
Can you show me an example of something that `(\S+)(\S*)` matches, and something that it doesn't match?

Reply:
You are right. I found I had no example which matches such a case. Rather, it turned out we were just lucky not to have, e.g., `import ansible.module_utils.util0, ansible.module_utils.util1, ansible.module_utils.util2, ...`
Let me rework on this pr. I'm setting WIP for now.

Reply:
@richm, fixed `(\S+)(\S*)`, which should have been `(\S+)(.*)` to capture as localname. And added the test cases.
```
    Adding test cases
      import ansible.module_utils.utils0  # noqa:E501
      import ansible.module_utils.utils1 as local1  # noqa:E501
    matching
      "(\bimport) (ansible\.module_utils\.)(\S+)(.*)(\s+#.+|.*)$"
    to test_lsr_role2collection.py.
```
Regarding `import ansible.module_utils.utils0, ansible.module_utils.utils1, ansible.module_utils.utils2`, I'd like to propose to give it up in the system roles. (See also https://issues.redhat.com/browse/RHELPLAN-54079?focusedCommentId=15375157&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-15375157)

Reply:
I think tools like black, flake8, pylint will reject `import ansible.module_utils.utils0, ansible.module_utils.utils1, ansible.module_utils.utils2` - if so, we definitely don't have to worry about that case - at any rate, I think we should not handle it

Reply:
You are right! Flake8 complains it:
```
import errno, functools, os, re, shlex, socket, subprocess, time, traceback
            ^
1     E401 multiple imports on one line
```
We are good. :)